### PR TITLE
Update NHD attribute table

### LINKED ISSUE #79: Move FHWA-NHDVariableList.xlsx to 1_fetch/in

### PR DESCRIPTION
**Pull request by jds485**
Updated the NHD attribute variable list to remove riparian buffer variables, and fix the 2011 land cover link.

Also moved this .xlsx file to 1_fetch/in, and added 2 file targets to track changes in the gages 2 and attribute spreadsheets

Closes #79

### CODE REVIEW

**Comment by jds485** [comment]
The new sb_table_reduced includes the NLCD land cover attributes that we want.
image

**Comment by cstillwellusgs** [comment]
Should Gages2.1_RefSiteList.xlsx be added to 01/fetch/in as well? Do we know when that spreadsheet will be made public?

**Comment by jds485** [comment]
> Should Gages2.1_RefSiteList.xlsx be added to 01/fetch/in as well? Do we know when that spreadsheet will be made public?

We can't git commit it until it's public, so for now I'm okay with leaving in the top directory. Let me ask Ken if he knows when it will be public

**Comment by cstillwellusgs** [comment]
Ahh I see - because this repo is public. Never thought about that!

**Comment by cstillwellusgs** [comment]
I'm getting this after `p1_sb_table_reduced_csv`, we okay with this?

> Variable not in sb_table_full: PPT7100_JAN
Variable not in sb_table_full: PPT7100_FEB
Variable not in sb_table_full: PPT7100_MAR
Variable not in sb_table_full: PPT7100_APR
Variable not in sb_table_full: PPT7100_MAY
Variable not in sb_table_full: PPT7100_JUN
Variable not in sb_table_full: PPT7100_JUL
Variable not in sb_table_full: PPT7100_AUG
Variable not in sb_table_full: PPT7100_SEP
Variable not in sb_table_full: PPT7100_OCT
Variable not in sb_table_full: PPT7100_NOV
Variable not in sb_table_full: PPT7100_DEC
Variable not in sb_table_full: SINUOSITY

**Comment by cstillwellusgs** [comment]
I'm getting this after `p1_sb_table_reduced_csv`, we okay with this?

> Variable not in sb_table_full: PPT7100_JAN
Variable not in sb_table_full: PPT7100_FEB
Variable not in sb_table_full: PPT7100_MAR
Variable not in sb_table_full: PPT7100_APR
Variable not in sb_table_full: PPT7100_MAY
Variable not in sb_table_full: PPT7100_JUN
Variable not in sb_table_full: PPT7100_JUL
Variable not in sb_table_full: PPT7100_AUG
Variable not in sb_table_full: PPT7100_SEP
Variable not in sb_table_full: PPT7100_OCT
Variable not in sb_table_full: PPT7100_NOV
Variable not in sb_table_full: PPT7100_DEC
Variable not in sb_table_full: SINUOSITY

**Comment by jds485** [comment]
Yes, those data links are added [manually](https://github.com/USGS-R/regional-hydrologic-forcings-ml/blob/main/1_fetch/src/get_sb_data.R#L121) in the same function. The links in the original table are incorrect / outdated. We probably should change how this `p1_sb_table_reduced_csv` target writes messages to say that it's adding the missing links.

**Comment by cstillwellusgs** [comment]
Right. On further inspection, none of the 1971-2000 monthly avg precip values were added though. If you look at the science base items that are manually added in the lines of code that follow the message, it goes here (example, the five others are the other combinations of ACC/TOT and temperature). Which begs the question, do we really need the monthly avgs from 1971-2000 if I calculate the monthly avgs in the next target (p1_monthly_weather_g2) based on the 1945-2015 data?

Sinuosity was successfully added, however. I'm not sure it was ever missing? It's SB item (57976a0ce4b021cadec97890) is in the original sb_table_full.csv doc. 

Long story short, we should probably update or remove the messages as it leads to some confusion.

**Review by cstillwellusgs — CHANGES_REQUESTED** [review]
Besides cleaning up the messages as stated in the above comment, should be good to go.

**Comment by cstillwellusgs** [comment]
This seems redundant? Is the first to track if the file changes, then the second to read in the file if it changes?

>   #file target for sciencebase variable list excel sheet
  tar_target(p1_sb_var_list_xlsx,
             sb_var_list_path,
             deployment = 'main',
             format = "file"
  ),
  #read in sciencebase variable list excel sheet
  tar_target(p1_sb_var_list,
             read_xlsx(path = p1_sb_var_list_xlsx, sheet = sb_var_sheet),
             deployment = 'main'
  ),

**Comment by jds485** [comment]
> This seems redundant? Is the first to track if the file changes, then the second to read in the file if it changes?

Yes

**Comment by cstillwellusgs** [comment]
targets still confuse me, apparently.

**Comment by jds485** [comment]
> Sinuosity was successfully added, however. I'm not sure it was ever missing? It's SB item (57976a0ce4b021cadec97890) is in the original sb_table_full.csv doc.

Good catch! I think the SB link is incorrect in the table. Let me update that and the message for Sinuosity should go away

**Comment by jds485** [comment]
> do we really need the monthly avgs from 1971-2000 if I calculate the monthly avgs in the next target (p1_monthly_weather_g2) based on the 1945-2015 data

No, I don't think we need the monthly values. Let me update the SB download table to remove the monthly values

**Comment by jds485** [comment]
@cstillwellusgs: I made the edits. Let me know if it's good to merge

**Review by cstillwellusgs — APPROVED** [review]
